### PR TITLE
Testing: Add VO Test framework

### PR DESCRIPTION
**Pull request by maany**
Here's what a VOTest run looks like:
https://github.com/maany/rucio/runs/5319690317?check_suite_focus=true 
Or
See the `VO specific tests` checks for this PR

New VOTests can be added by configuring the build matrix for votests. 
For example, if we want to add tests for belleii, we would add the following section in the `matrix_policy_pacakge_tests.yml`

```
belleii:
  installation_cmd: ""
  config_overrides:
    permission: belleii
    extract_scope:  belleii
    schema: belleii
    lfn2pfn_algorithm_default: "belleii"
  tests:
    allow:
      - rucio_tests
      - rucio_tests/belleii
    deny:
      - rucio_tests/belleii/test_belleii.py
  rdbms: 
    - postgres12
  python:
    - "3.6"
  dists:
    - centos7
  image_identifier: 
    - belleii
```

This PR also introduces composite actions i.e. re-usable actions for building images and running tests across multiple workflows ( integration, votest, autotest). Currently, only the votests use this composite workflow and we can adapt the rest of the workflows to use it once this PR has been merged.
See the file: [.github/actions/build-images-and-run-tests/action.yml](https://github.com/rucio/rucio/pull/5286/files#diff-1a69b8cbd1dc57f38635b35d01dcfd4e2a2217e43436c7c0641de281fa39cb8a)

### CODE REVIEW

File: tools/test/votest_helper.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                        type=lambda p: Path(p).absolute(),
+                        default=Path(Path(__file__).parent.parent.parent / "etc" / "rucio.cfg").absolute(),
+                        help='the path to rucio.cfg',
+                        )
+    parser.add_argument('--vo', type=str, default='all', required=False)
+    parser.add_argument('-v', action='store_true',
+                        required=False,
+                        help="Verbose mode to show logged errors")
+    output_type = parser.add_mutually_exclusive_group(required=False)
+    output_type.add_argument('-i', required=False,
```
Comment (joeldierkes) [inline tools/test/votest_helper.py]: This name is not descriptive. How about `--return-policy-package-installation-command`, or `--policy-package-installation-command`?

File: tools/test/votest_helper.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+                             action='store_true',
+                             help="return the policy package installation command for given vo."
+                             )
+    output_type.add_argument('-c', required=False,
+                             action='store_true',
+                             help="return the rucio config section for the given vo.")
```
Comment (joeldierkes) [inline tools/test/votest_helper.py:176]: Same as above. `--get-vo-config`?

File: tools/test/votest_helper.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    output_type.add_argument('-t', required=False,
+                             action='store_true',
+                             help="return the tests that will be run for the given vo.")
```
Comment (joeldierkes) [inline tools/test/votest_helper.py:179]: `--return-test`, `--return-vo-test`?

File: tools/test/votest_helper.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+import configparser
+from pathlib import Path
+import glob
+import itertools
+import yaml
+
+logger = logging.getLogger(__name__)
+
+
+def validate(func):
```
Comment (joeldierkes) [inline tools/test/votest_helper.py:35]: A comment would be good.

File: etc/docker/test/matrix_policy_package_tests.yml [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  installation_cmd: ""
+  config_overrides: 
+    permission: atlas
+    extract_scope:  atlas
+    schema: atlas
+    lfn2pfn_algorithm_default: "hash"
+  tests:
+    allow:
+      - rucio_tests
+    deny:
```
Comment (joeldierkes) [inline etc/docker/test/matrix_policy_package_tests.yml:11]: Every time a new test file is created, it needs to be added here. This seems to add a constraint nobody will remember in the future. Can we change this to only specify the allowed ones?

**Review by joeldierkes — CHANGES_REQUESTED** [review]

File: etc/docker/test/matrix_policy_package_tests.yml (hunk 5 above)
Comment (joeldierkes) [reply etc/docker/test/matrix_policy_package_tests.yml:11]: These test files will be migrated to use the newly created factories.

File: tools/test/votest_helper.py [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+                        help='the path to matrix_policy_package_tests.yml',
+                        )
+    parser.add_argument('--rucio-cfg',
+                        metavar="rucio",
+                        type=lambda p: Path(p).absolute(),
+                        default=Path(Path(__file__).parent.parent.parent / "etc" / "rucio.cfg").absolute(),
+                        help='the path to rucio.cfg',
+                        )
+    parser.add_argument('--vo', type=str, default='all', required=False)
+    parser.add_argument('-v', action='store_true',
```
Comment (joeldierkes) [inline tools/test/votest_helper.py]: `--verbose`

**Review by bari12 — APPROVED** [review]

**Pull request merged**